js/src/iaa.js

```diff
@@ -58,6 +58,39 @@ class IntervalAgreementApproach {
     );
     return top / bottom;
   }
+
+  /**
+   * Calculating mean of maxima is similar to centroid in that it will need to 
+   * sample linearly spaced values across a range (or ranges!), but then it will average all sampled values
+   */
+  get mean_of_maxima() {
+    const smallestValue = Math.min(...this.intervals.singletonKeys);
+    const largestValue = Math.max(...this.intervals.singletonKeys);
+    const values = linspace(
+      [smallestValue, largestValue],
+      (largestValue - smallestValue) * 10 + 1
+    )
+    const sum = values.reduce((a, b) => a + b, 0)
+    mean = sum/values.length
+    
+    return mean;
+  }
+
+  /**
+   * sum all minimum values and all maximum values then divide 
+   * by the count of all summed values (i.e. sum of the counts of min and max)
+   */
+  get mean_of_midpoints() {
+    const minValues = this.intervals.singletonKeys.filter(val=> val == Math.min(...this.intervals.singletonKeys))
+    const maxValues = this.intervals.singletonKeys.filter(val=> val == Math.max(...this.intervals.singletonKeys))
```

Comment:
I don't think singletonKeys works for us here:

Imagine a set of intervals: `[1,5], [1,3], [2,5]`

All the min and max values in that set are `1,1,2,3,5,5` but the *distinct* values (what I think `singletonKeys` would get) are `1,2,3,5`.

Let's try and take the mean of each set of values:
- `1,1,2,3,5,5` -> `17/6` = `2.83`
- `1,2,3,5` -> `11/4` = `2.75`

They don't match, so it's obviously important that we include how many times we have the same value, so I think just summing `singletonKeys` is wrong. Instead we need each key, times the number of participants that selected it, and then the divisor needs to include the correct count as well.